### PR TITLE
Modifying the plot option control in Line plot dialog. >Describe>Specific Tables/Graphs>Line/Smooth/Dumbbell/Slope

### PR DESCRIPTION
**Pull request by Fidel365**
Fixes (partially) #7556 
@N-thony , @lloyddewit , this PR is ready for review

### CODE REVIEW

**Comment by lloyddewit** [comment]
@N-thony Please could you review?
@rdstern Please could you test?
Thanks

**Comment by rdstern** [comment]
@Fidel365 this is a good start.  But there is quite a lot to add.  When I pressed on point then that should be added to the list.  Similarly for peaks and valleys.  Perhaps for valleys the control could be better lined up at the same time.
When on smooth, and then ask for the line, then that should be added.
By the way, at least initially, I am quite relaxed about whether an option is enabled - as you are trying to do.  If initially they are there, that's the important part.

File: instat/dlgLinePlot.vb [hunk 1]
```diff
@@ -730,4 +686,63 @@ Public Class dlgLinePlot
...
+        TestOkEnabled()
+    End Sub
+
+    Private Sub AllControl_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrVariablesAsFactorForLinePlot.ControlContentsChanged, ucrSave.ControlContentsChanged, ucrReceiverXEnd.ControlContentsChanged, ucrReceiverX.ControlContentsChanged, ucrReceiverSlopeY.ControlContentsChanged, ucrReceiverSlopeX.ControlContentsChanged, ucrReceiverSlopeColour.ControlContentsChanged, ucrFactorOptionalReceiver.ControlContentsChanged
+
+    End Sub
+
+    Private Sub UcrVariablesAsFactor_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrVariablesAsFactorForLinePlot.ControlValueChanged
+
+    End Sub
```
Comment (N-thony) [inline instat/dlgLinePlot.vb:747]: Can you remove the empty sub?

File: instat/dlgLinePlot.vb [hunk 2]
```diff
@@ -557,70 +557,25 @@ Public Class dlgLinePlot
...
         If ucrVariablesAsFactorForLinePlot.bSingleVariable Then
-            cmdLinePathStepSmoothOptions.Enabled = True
+            'cmdLinePathStepSmoothOptions.Enabled = True
             cmdOptions.Enabled = True
         Else
-            cmdLinePathStepSmoothOptions.Enabled = False
+            'cmdLinePathStepSmoothOptions.Enabled = False
             cmdOptions.Enabled = False
         End If
     End Sub
```
Comment (N-thony) [inline instat/dlgLinePlot.vb:566]: we can get rid of this since it called on line 569

File: instat/dlgLinePlot.vb [hunk 3]
```diff
@@ -557,70 +557,25 @@ Public Class dlgLinePlot
...
-            ElseIf clsParam.strArgumentName = "xend" Then
-                ucrReceiverXEnd.Add(clsParam.strArgumentValue)
-            End If
-        Next
-        ucrInputMethod.SetRCode(clsOptionsFunction, bReset)
-        TestOkEnabled()
-    End Sub
-
     Private Sub UcrVariablesAsFactor_ControlValueChanged() Handles ucrVariablesAsFactorForLinePlot.ControlValueChanged
         TempOptionsDisabledInMultipleVariablesCase()
```
Comment (N-thony) [inline instat/dlgLinePlot.vb:569]: then we replace this by
cmdLinePathStepSmoothOptions.Enabled = ucrVariablesAsFactorForLinePlot.bSingleVariable

File: instat/dlgLinePlot.vb [hunk 4]
```diff
@@ -643,7 +598,7 @@ Public Class dlgLinePlot
 
     Private Sub SetGraphPrefixAndRcommand()
         ucrVariablesAsFactorForLinePlot.SetMeAsReceiver()
-        cmdLinePathStepSmoothOptions.Visible = True
+        'cmdLinePathStepSmoothOptions.Visible = True
```
Comment (N-thony) [inline instat/dlgLinePlot.vb:601]: We can remove this if not needed?

File: instat/dlgLinePlot.vb [hunk 5]
```diff
@@ -674,12 +629,12 @@ Public Class dlgLinePlot
             clsBaseOperator.AddParameter("dumbbellplot", clsRFunctionParameter:=clsDumbbellFunction, iPosition:=1)
         ElseIf rdoSlope.Checked Then
             ucrReceiverSlopeY.SetMeAsReceiver()
-            cmdLinePathStepSmoothOptions.Visible = False
+            'cmdLinePathStepSmoothOptions.Visible = False
```
Comment (N-thony) [inline instat/dlgLinePlot.vb:632]: here too

**Review by N-thony — CHANGES_REQUESTED** [review]
@Fidel365 can you resolve my first comments above? Thanks

**Comment by rdstern** [comment]
@N-thony this still has conflicts.  I wonder who should take this over from @Fidel365 who I assume is no longer doing this now?

**Comment by Fidel365** [comment]
I will be working on this and it will be ready before the end of tomorrow

**Comment by derekagorhom** [comment]
@africanmathsinitiative/developers i taking over this PR

**Comment by lloyddewit** [comment]
Replaced by PR #8056

**Pull request closed without merging**